std/range/package.d

```diff
@@ -7125,9 +7131,57 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Chunks* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
+        @property auto front() { return impl.r.front; }
+        void popFront()
+        {
+            assert(impl.curSizeLeft > 0 && !impl.r.empty);
+            impl.curSizeLeft--;
+            impl.r.popFront();
+        }
+    }
+
+    private Source r;
+    private size_t chunkSize;
+    private size_t curSizeLeft;
+    bool empty;
+
+    this(Source _r, size_t _chunkSize)
+    {
+        r = _r;
+        chunkSize = _chunkSize;
+        curSizeLeft = chunkSize;
+        empty = r.empty;
+    }
+
+    @property Chunk front() { return Chunk(&this); }
```

Comment:
Haha, just saw this comment... but this is already causing a breakage when compiled with `-dip25 -dip1000` because the new `isInputRange` tests `front` with `typeof((R r) => r.front)`, which *will* escape the reference to `this`, and causes it to reject this as being a range.

I suppose I should turn this into a class rather than a struct... in the original non-Phobos version of this in my own code, the wrapping `Chunks` is always allocated on the heap.  The problem with `-dip25 -dip1000` still exists, though.